test/testutil.py

```
@@ -147,3 +149,270 @@ def create_filesystem_customizations(rootfs: str):
+    _email: str = "bib-tests@redhat.com"
+    _key_length: str = "3072"
+    _key_params: str = _key_params_tmpl.format(key_length=_key_length, email=_email)
+
+    @property
+    def base_dir(self) -> str:
+        return self._base_dir
+
+    @base_dir.setter
+    def base_dir(self, base_dir: str) -> None:
+        self._base_dir = base_dir
+        self._home_dir = f"{base_dir}/.gnupg"
+        self._pub_key_file = f"{base_dir}/GPG-KEY-bib-tests"
+
+    @property
+    def home_dir(self) -> str:
+        return self._home_dir
+
+    @home_dir.setter
+    def home_dir(self, home_dir: str) -> None:
+        self._home_dir = home_dir
+
+    @property
+    def pub_key_file(self) -> str:
+        return self._pub_key_file
+
+    @pub_key_file.setter
+    def pub_key_file(self, file: str) -> None:
+        self._pub_key_file = file
+
+    @property
+    def email(self) -> str:
+        return self._email
+
+    @email.setter
+    def email(self, email: str) -> None:
+        self._email = email
+        self._key_params = self._key_params_tmpl.format(
+            email=self._email,
+            key_length=self._key_length
+        )
+
+    @property
+    def key_length(self) -> str:
+        return self._key_length
+
+    @key_length.setter
+    def key_length(self, length: str) -> None:
+        self._key_length = length
+        self._key_params = self._key_params_tmpl.format(
+            email=self._email,
+            key_length=self._key_length
+        )
+
+    @property
+    def key_params(self) -> str:
+        return self._key_params
+
+    @key_params.setter
+    def key_params(self, params: str) -> None:
+        self._key_params = params
+
+
+def gpg_gen_key(gpg_conf: GPGConf):
+    if os.path.exists(gpg_conf.home_dir):
+        return
+
+    os.makedirs(gpg_conf.home_dir, mode=0o700, exist_ok=False)
+
+    subprocess.run(
+        ["gpg", "--gen-key", "--batch"],
+        check=True, capture_output=True,
```

I wonder if we want `capture_output=True` here. By default pytest will hide the output but if anything goes wrong this catpure_output=True will mean the errors are not visible by default in the exception (subprocess.CalledProcessError). Maybe just not capture here and below (unless I miss something?)